### PR TITLE
Closes #907: Setup production box for https://telescope.cdot.systems

### LINKED ISSUE #907: Setup production box for https://telescope.cdot.systems
We need to get our production box created before we can ship 1.0.  It will be very similar to staging (ideally as similar as possible, so testing one means we've tested the other).  However, there will be differences:

- tweak any settings in our `.env`
- separate `.env` and other config files will need to be created
- modify our auto deployment so it only happens when releases get tagged vs. merges to master
- setup SSO with real Seneca authentication system

I'm sure there is more, but this is what comes to mind right now.

### PR DESCRIPTION
**Pull request by manekenpix**
Co-authored-by: James Inkster <James.inkster@live.ca>

## Issue This PR Addresses

Closes #907 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

This adds settings for the production box. I'll keep this PR as `draft` until we can add Seneca SSO information for production.

- [x] Tweak any settings in our `.env`
- [x] Separate .env and other config files will need to be created
- [ ] Modify our auto deployment so it only happens when releases get tagged vs. merges to master
 - This will be done in #957 
- [ ] Setup SSO with real Seneca authentication system
 - This will be added to this PR once we get the information needed from Seneca.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by humphd** [comment]
I assume the `docker-compose-production.yml` is coming later?

File: env.production [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+# FEED_URL url used to access feed list
+FEED_URL=https://wiki.cdot.senecacollege.ca/wiki/Planet_CDOT_Feed_List
+# Milliseconds to wait after attempting to fetch the feed list when the server is not available
+FEED_URL_INTERVAL_MS=30000
+
+# Redis Server info, password may be optional (e.g., leave empty if you don't set one)
+REDIS_URL=redis://127.0.0.1
+REDIS_PORT=6379
+REDIS_PASSWORD=
```
Comment (humphd) [inline env.production:25]: Remove any variables we aren't going to set.

File: env.production [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+# Elasticsearch info
+ELASTIC_URL=http://127.0.0.1
+ELASTIC_PORT=9200
+# Max number of results per query
+ELASTIC_MAX_RESULTS=50
+# Delay to check connectivity with Elasticsearch in ms
+ELASTIC_DELAY_MS=10000
+
+# The Single Sign On (SSO) login service URL
+SSO_LOGIN_URL=
```
Comment (humphd) [inline env.production:39]: I better email Mehrdad to get this started...

File: env.production [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+SECRET=
+
+# ADMINISTRATORS is a list (space delimited) of users who have administrator
+# rights. Use the user's nameID (user2@example.com) or hashed version of
+# nameID (2b3b2b9ce8).  Either will work.
+ADMINISTRATORS=david.humphrey@senecacollege.ca chris.tyler@senecacollege.ca
+
+# BLOG_INACTIVE_TIME is the period (days) of inactivity
+# before a blog will be considered redlisted
+BLOG_INACTIVE_TIME=360
```
Comment (humphd) [inline env.production:67]: This can go, we don't use it.  I'll file a bug to remove from other files.

File: env.production [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+# Feed job queue delay(ms)
+FEED_QUEUE_DELAY_MS=60000
+
+# Number of concurrent worker processors to run. Use * if you want to run
+# one per CPU. Use a number if you want to set it manually, up to a max
+# of the CPU count.  If not set, we'll assume 1.
+FEED_QUEUE_PARALLEL_WORKERS=1
+
+# Max number of posts per page
+MAX_POSTS_PER_PAGE=50
```
Comment (humphd) [inline env.production:85]: I think we should reduce this to 30 again.

**Comment by manekenpix** [comment]
@humphd I think `docker-compose-staging.yml` and `docker-compose-production.yml`would be identical, so I'm not sure if we should have 2 identical files with different names or just rename `docker-compose-staging.yml` to something that'd indicate that it's used in both staging and production.

**Comment by humphd** [comment]
OK, maybe we should call it `docker-compose-production.yml`, and we'll use that on staging and production.

File: docker-compose-production.yml [hunk 5]
```diff
@@ -4,7 +4,7 @@ version: '3'
 
 services:
   telescope:
-    container_name: telescope_staging
+    container_name: telescope_production
```
Comment (humphd) [inline docker-compose-production.yml:7]: Do we need to update https://github.com/Seneca-CDOT/telescope/blob/d080857cb959687b0182b11d7e630b540ac24dee/nginx-staging.conf to use this new name?  Or is that separate?

Comment (manekenpix) [reply docker-compose-production.yml:7]: Oh, yes, my bad.

**Comment by manekenpix** [comment]
I just realized we do need 2 separate `docker-compose` files

File: env.production (hunk 2 above)
Comment (humphd) [reply env.production:35]: NOTE: this has been done.  Waiting to hear back.

File: tools/autodeployment/deploy_stage.sh [hunk 6]
```diff
@@ -3,7 +3,7 @@
 # Shutdown
 cd ../telescope
 
-docker-compose -f docker-compose-staging.yml down
+docker-compose -f docker-compose-production.yml down
```
Comment (humphd) [inline tools/autodeployment/deploy_stage.sh:6]: In @c3ho's PR, we can deal with this, but calling this file `deploy_stage.sh` is odd if we use `-production` here.

Comment (manekenpix) [reply tools/autodeployment/deploy_stage.sh:6]: I think we can try to merge this one first and then @c3ho can rebase and change the name of the file and make the necessary adjustments.

Comment (c3ho) [reply tools/autodeployment/deploy_stage.sh:6]: I'm fine with that

**Pull request merged**